solidity/contracts/Coordinator.sol

```diff
@@ -28,16 +27,45 @@ contract Coordinator {
     uint256 _payment,
     uint256 _expiration,
     address[] _oracles,
+    uint8[] _vs,
+    bytes32[] _rs,
+    bytes32[] _ss,
     bytes32 _requestDigest
   ) public
   {
-    bytes32 id = getId(_payment, _expiration, _oracles, _requestDigest);
+    require(
+      (_oracles.length == _vs.length) && (_vs.length == _rs.length) && (_rs.length == _ss.length),
+      "Must pass in as many signatures as oracles");
+
+    bytes32 _serviceAgreementID = getId(_payment, _expiration, _oracles, _requestDigest);
 
-    serviceAgreements[id] = ServiceAgreement(
+    for (uint i = 0; i < _oracles.length; i++) {
+      address signer = getOracleAddressFromSASignature(_serviceAgreementID, _vs[i], _rs[i], _ss[i]);
+      require(
+        _oracles[i] == signer,
+        "!!! oracle is not the signer: TODO: can it do string interpolation of the addresses???"
```

Comment:
I think I added this awesome message :) But we should update it before coming out of WIP. How do you feel about something like: `The given oracles did not sign the message`?

We could concat the index to make the message more useful but is it worth the extra gas cost?